conan/tools/files/cpp_package.py

```diff
@@ -47,7 +47,11 @@ def package_info(self, conanfile):
             conanfile.cpp_info.components[cname].libs = component.libs
             conanfile.cpp_info.components[cname].requires = component.requires
             for generator, gname in component.names.items():
-                conanfile.cpp_info.components[cname].set_property("cmake_target_name", gname, generator)
+                # TODO: probably we want to revisit all this stuff with the new absolute
+                #  target names via set_property
+                conanfile.cpp_info.components[cname].set_property("cmake_target_name",
+                                                                  "{}::{}".format(conanfile.cpp_info.name, gname),
```

Comment:
Is there a way we could do this without relying on ``cpp_info.name``? Because this will break merging to ``develop2``

This ``cpp_package`` is a new feature, should be ok if doing only the new thing, or supporting only CMakeDeps.